File: backend_infrastructure/src/main/java/com/bankfy/bank_meet/domain/models/Cuenta.java
package com.bankfy.bank_meet.domain.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.EqualsAndHashCode;
import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "cuenta")
public class Cuenta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false, updatable = false)
    private String numeroCuenta;

    @Column(nullable = false)
    private String tipoCuenta;

    @Column(nullable = false)
    private BigDecimal saldoInicial;

    @Column(nullable = false)
    private Boolean estado;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "cliente_id", nullable = false, updatable = false)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private Cliente cliente;
}

File: frontend_infrastructure/src/app/domain/models/cuenta/cuenta.ts
export interface Cuenta {
  id?: number;
  numeroCuenta: string;
  tipoCuenta: 'Ahorros' | 'Corriente' | 'Ahorro Programado' | 'Poliza';
  saldoInicial: number;
  estado: boolean;
  clienteId: number;
  nombreCliente?: string;
  identificacionCliente?: string;
}
